api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3505,7 +3509,13 @@ private void putObject(String bucketName, String objectName, Long size, Object d
         }
       }
 
-      String etag = putObject(bucketName, objectName, expectedReadSize, data, uploadId, partNumber, null);
+      // In multi-part uploads, Set encryption headers in the case of SSE-C.
+      Map<String, String> encryptionHeaders = new HashMap<>();
+      if (sse != null && sse.getType() == ServerSideEncryption.Type.SSE_C) {
+        sse.marshal(encryptionHeaders);
+      }
+      
+      String etag = putObject(bucketName, objectName, expectedReadSize, data, uploadId, partNumber, encryptionHeaders);
```

Comment:
@kannappanr could you add functional tests to verify multipart upload encryption

Reply:
done